### ISSUE TITLE
NVDA Fails To Render Browser Pages After Updating

### ISSUE THREAD

**Issue by Elshara** [nvaccess/nvda#14890]
### Steps to reproduce:
1. Open a browser window. Google Chrome, Firefox, Microsoft Edge/internet explorer and navigate to any web document, local or online.
2. Install NVDA Alpha/Next for a faster reproduction scenario. Or alternatively, use the last two releases of stable NVDA as a fallback.
3. Install the available package as normal. Note this happens regardless of addons, and whether or not they're enabled, disabled or you have no addons installed. It also happens on portable and stationary copies.
4. After NVDA restarts, navigate back to your opened web document and try to read it. Notice there is quite a bit of freezing lag here as NVDA fails to render in page navigation support.
### Actual behavior:
NVDA fails to render browser support after it updates to any version. And the only way to fix this, is to restart the computer.

### Expected behavior:
NVDA to serve web pages as normal, even if you have to renavigate to where you left off on the web page when the update to NVDA version was processed as its position as a virtual cursor in the document would lose its memory based bookmark.

### NVDA logs, crash dumps and other attachments:
This I'm not clear on as I cannot access them using any other screen reader without restarting the computer.
### System configuration
Windows 11.
#### NVDA installed/portable/running from source:
NVDA Alpha 28166
#### NVDA version:
2023.2
#### Windows version:
Home
#### Name and version of other software in use when reproducing the issue:
Chrome stable, Firefox Stable
#### Other information about your system:
Ram 64 gig, CPU Ryzen Threadripper 2920w, GPU NVIDIA RTX2060
#### Does the issue still occur after restarting your computer?
no
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no/can't
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
n/a

**Comment by XLTechie** [comment]
@Elshara

>     Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
> n/a

Why do you say "n/a" for this, when it seems very much applicable?

The tool is designed exactly to solve such problems as the one you described.
Or, at least, it is a first debugging step to perform when browsers, especially 
many browsers, start behaving badly.

Please run that tool, reboot, and determine if the problem happens again.

Thanks

**Comment by Elshara** [comment]
Running this tool should not be an extra step required for every NVDA user immediately after it updates. N/A or not/available or not/applicable is the same everywhere. It means, it doesn't apply to this issue specifically because the issue does not address a registration. It addresses something completely different. And I have already stated restarting the computer is the only way to resolve this issue. So the problem is with NVDA's side immediately after updating. The second NVDA starts up after it already restarts to apply the latest update, this issue occurs. And the computer is too laggy to do anything else accept restart the computer as no other troubleshooting steps are available if a browser page is open. I strongly suspect that not all child threads of NVDA's processes are quit when NVDA relaunches itself after it updates. Since this issue goes away after a restart, I highly suggest looking into what is left in memory when NVDA is updated and see if the browser is not responding after trying to communicate with information NVDA is working with that is likewise not responding as the thread ID is already expired. A simple fix of completely turning off NVDA completely on the system would fix this while it is updating so threads can properly shut down. It's the only logical solution I can think of, since it is in actual effect a temporary problem either to do with memory, caching, or extra running processes of NVDA under task manager which overlap with one another during updates and immediately after NVDA is first restarted.

**Comment by XLTechie** [comment]
> Running this tool should not be an extra step required for every NVDA user immediately after it updates.

You are, so far, the only NVDA user reporting this problem, as opposed to "every 
NVDA user".

In the STR, you imply that the last two stable versions of NVDA experience this 
issue at least to some extent, which strongly suggests we would have heard of it 
before now, if it was generalizable.

Unfortunately, I do not have Windows 11 in order to test this.

That said, restarting the machine after an installed NVDA upgrade, is the 
recommended course of action anyway; and after up/downgrading NVDA a few times 
without restarting, you will inevitably run into difficulties with COM proxies.

You also indicate that it happens after updating a portable copy, or that's what 
I believe you are saying.
Does the same thing happen if you just restart a portable copy (NVDA+q, 
choose restart, press enter)

Does the same thing happen if you update a copy (portable or installed), then 
completely shut down NVDA, wait a moment, then start it again?

**Comment by seanbudd** [comment]
Please restart your device after updating NVDA, like instructed on the download page.
I've opened https://github.com/nvaccess/nvda/issues/14893 to be more explicit about this requirement.

**Comment by seanbudd** [comment]
Closing this as won't fix
